sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBundleGasPerBlockTransactionSelector.java

```diff
@@ -60,15 +49,67 @@ public TransactionSelectionResult evaluateTransactionPostProcessing(
       final TransactionEvaluationContext txContext,
       final TransactionProcessingResult transactionProcessingResult) {
 
-    // short circuit if we are not a PendingBundleTx
-    if (!(txContext.getPendingTransaction() instanceof LineaSendBundle.PendingBundleTx)) {
-      return SELECTED;
-    }
+    if (txContext.getPendingTransaction()
+        instanceof BundlePoolService.TransactionBundle.PendingBundleTx pendingBundleTx) {
+
+      final long gasUsedByTransaction =
+          transactionProcessingResult.getEstimateGasUsedByTransaction();
 
-    final long newCumulativeBundleGas =
-        getWorkingState() + txContext.getPendingTransaction().getTransaction().getGasLimit();
-    setWorkingState(newCumulativeBundleGas);
+      final long currentBundleGasUsed =
+          pendingBundleTx.isBundleStart() ? 0L : getWorkingState().currentBundleGasUsed();
+      final long newCurrentBundleGasUsed = currentBundleGasUsed + gasUsedByTransaction;
 
+      final long cumulativeBlockBundleGasUsed = getWorkingState().cumulativeBlockBundleGasUsed();
+      final long newCumulativeBlockBundleGasUsed =
+          cumulativeBlockBundleGasUsed + gasUsedByTransaction;
+
+      setWorkingState(
+          new BundleGasTracker(newCumulativeBlockBundleGasUsed, newCurrentBundleGasUsed));
+
+      if (newCurrentBundleGasUsed > maxBundleGasPerBlock) {
+        log.atTrace()
+            .setMessage(
+                "Not selecting bundle transaction {} since the current gas used by the bundle is greater than the max {};"
+                    + " gas used by tx {} + gas already used by the bundle {} = {}")
+            .addArgument(pendingBundleTx::toTraceLog)
+            .addArgument(maxBundleGasPerBlock)
+            .addArgument(gasUsedByTransaction)
+            .addArgument(currentBundleGasUsed)
+            .addArgument(newCurrentBundleGasUsed)
+            .log();
+        return BUNDLE_GAS_EXCEEDS_MAX_BUNDLE_BLOCK_GAS;
+      }
+
+      if (newCumulativeBlockBundleGasUsed > maxBundleGasPerBlock) {
+        log.atTrace()
+            .setMessage(
+                "Not selecting bundle transaction {} since the current gas used by the bundle is greater than the max {};"
```

Comment:
```suggestion
                "Not selecting bundle transaction {} since the cumulative gas used by bundles in this block is greater than the max {};"
```

Reply:
unsure about this log message but I think it has to be about the cumulative usage

Reply:
good catch, probably a wrong copy & paste